server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -807,6 +808,18 @@ protected Node(
                 remoteIndexPathUploader = null;
                 remoteClusterStateCleanupManager = null;
             }
+            final RemoteStorePinnedTimestampService remoteStorePinnedTimestampService;
+            if (isRemoteStoreAttributePresent(settings)) {
```

Comment:
Should use `isRemoteDataAttributePresent` here instead since this is required for remote data store.